src/gfn/gflownet/base.py

```diff
@@ -63,15 +77,24 @@ class PFBasedGFlowNet(GFlowNet[TrainingSampleType]):
         pb: GFNModule
     """
 
-    def __init__(self, pf: GFNModule, pb: GFNModule, on_policy: bool = False):
+    def __init__(self, pf: GFNModule, pb: GFNModule, off_policy: bool):
         super().__init__()
         self.pf = pf
         self.pb = pb
-        self.on_policy = on_policy
+        self.off_policy = off_policy
 
-    def sample_trajectories(self, env: Env, n_samples: int) -> Trajectories:
+    def sample_trajectories(
+        self, env: Env, n_samples: int, sample_off_policy: bool, **policy_kwargs
```

Comment:
Because you might want to train off policy, but then sample later on policy, was my thinking.

It's easier for the user to reason about what they're doing if, when they call the sampler, they explicitly ask for on/off policy behaviour, instead of needing to check which "mode" the GFN is in.